github4s/src/main/scala/github4s/algebras/Repositories.scala

```diff
@@ -271,6 +271,26 @@ trait Repositories[F[_]] {
       headers: Map[String, String] = Map()
   ): F[GHResponse[List[User]]]
 
+  /**
+   * Get whether a user is a repository collaborator
+   *
+   * For organization-owned repositories, the list of collaborators includes outside collaborators,
+   * organization members that are direct collaborators, organization members with access through team memberships,
+   * organization members with access through default organization permissions, and organization owners.
+   *
+   * @param owner of the repo
+   * @param repo name of the repo
+   * @param username Github username
+   * @param headers optional user headers to include in the request
+   * @return a unit GHResponse
+   */
+  def userIsCollaborator(
```

Comment:
I think the return type is a bit confusing with respect to the method name. I would expect it to return `F[Boolean]`, but all the existing algebra methods I checked return a `GHResponse`, so that appears to be a design rule of the library.

@BenFradet WDYT?

Reply:
I did consider this but decided to maintain the `GHResponse` for two reasons:

1. Maintaining consistency with the library (as you mentioned)
2. Giving the user flexibility to handle other errors as they wish